### DISCUSSION TITLE
Updating Capacitor

### DISCUSSION THREAD

**Discussion by lorenzpfei** [ionic-team/capacitor#3123] [Q&A]
> 💊   Capacitor Doctor  💊
> 
> Latest Dependencies:
> 
>   @capacitor/cli: 2.2.0
> 
>   @capacitor/core: 2.2.0
> 
>   @capacitor/android: 2.2.0
> 
>   @capacitor/ios: 2.2.0
> 
> Installed Dependencies:
> 
>   @capacitor/android not installed
> 
>   @capacitor/cli 1.5.2
> 
>   @capacitor/ios 2.2.0
> 
>   @capacitor/core 2.2.0
> 
> 
> [success] Android looking great! 👌
>   Found 0 Capacitor plugins for ios:
> [success] iOS looking great! 👌

If I update my ionic project to capacitor >= 2.0.0 I always get an error in xcode:

> error: ...project/node_modules/@capacitor/ios/ios/CapacitorCordova/CapacitorCordova/Classes/Public/AppDelegate.h: No such file or directory (in target 'CapacitorCordova' from project 'Pods')

> ...project/node_modules/@capacitor/ios/ios/CapacitorCordova/CapacitorCordova/Classes/Public/AppDelegate.h: No such file or directory

If I move back to a lower version It works fine but I need the newer version to set specific CameraOptions.
What to do?
Thanks in advance.

**Comment by bipoza** [comment]
Have you tried to also update @capacitor/cli to version 2.2.0?

**My procedure is usually the following:**
1. Update all Capacitor packages
```
npm install @capacitor/ios@latest @capacitor/cli@latest @capacitor/core@latest @capacitor/android@latest --save
```
2. Sync the projects
```
npx cap sync ios
or 
npx cap sync android
```
3. Make the necessary changes in the native project
[IOS Project](https://capacitor.ionicframework.com/docs/ios/updating#from-1-5-1-to-2-0-0)
[ANDROID Project](https://capacitor.ionicframework.com/docs/android/updating#from-1-5-1-to-2-0-0)

4. Just in case, it never hurts to do a "Clean Build Folder"

<br>
Hope it helps, sometimes we get stuck with this kind of tasks ;)

**Reply by lorenzpfei** [reply]
Did exactly what you said but the same error occurs. If you look at the first error: Maybe I can change something in the Podfile?

**Comment by jcesarmobile** [comment]
Do you have `install! 'cocoapods', :disable_input_output_paths => true` in the Podfile?

**Reply by lorenzpfei** [reply]
> Do you have `install! 'cocoapods', :disable_input_output_paths => true` in the Podfile?

Yes. It's true.
My Podfile:

> platform :ios, '11.0'
> use_frameworks!
> 
> # workaround to avoid Xcode 10 caching of Pods that requires
> # Product -> Clean Build Folder after new Cordova plugins installed
> # Requires CocoaPods 1.6 or newer
> install! 'cocoapods', :disable_input_output_paths => true
> 
> def capacitor_pods
>   # Automatic Capacitor Pod dependencies, do not delete
>   pod 'Capacitor', :path => '../../node_modules/@capacitor/ios'
>   pod 'CapacitorCordova', :path => '../../node_modules/@capacitor/ios'
>   
>   # Do not delete
> end
> 
> target 'projname' do
>   capacitor_pods
>   # Add your Pods here
> end


My package.json:

> "dependencies": {
    "@angular/common": "~8.2.14",
    "@angular/core": "~8.2.14",
    "@angular/forms": "~8.2.14",
    "@angular/platform-browser": "~8.2.14",
    "@angular/platform-browser-dynamic": "~8.2.14",
    "@angular/router": "~8.2.14",
    "@babel/runtime": "^7.9.2",
    "@capacitor/android": "^2.2.0",
    "@capacitor/core": "^2.2.0",
    "@capacitor/ios": "^2.2.0",
    "@ionic-native/core": "^5.0.0",
    "@ionic-native/splash-screen": "^5.0.0",
    "@ionic-native/status-bar": "^5.0.0",
    "@ionic/angular": "^5.0.0",
    "core-js": "^2.5.4",
    "ngx-quill": "^10.0.1",
    "quill": "^1.3.7",
    "rxjs": "~6.5.1",
    "tslib": "^1.9.0",
    "zone.js": "~0.9.1"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^0.803.24",
    "@angular/cli": "8.3.26",
    "@angular/compiler": "~8.2.14",
    "@angular/compiler-cli": "~8.2.14",
    "@angular/language-service": "~8.2.14",
    "@babel/compat-data": "7.8.0",
    "@babel/preset-env": "7.8.0",
    "@capacitor/cli": "^2.2.0",
    "@ionic/angular-toolkit": "^2.1.1",
    "@ionic/lab": "3.1.6",
    "@types/jasmine": "~3.3.8",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "babel-loader": "8.1.0",
    "codelyzer": "^5.0.0",
    "jasmine-core": "~3.4.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.1.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.0",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.15.0",
    "typescript": "~3.4.3"
  },
  "resolutions": {
    "@babel/preset-env": "^7.8.0"
  },
